File: DaS.ScriptLib.CS/Game/Data/Helpers/EventFlagHelper.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using DaS.ScriptLib.LuaScripting;

namespace DaS.ScriptLib.Game.Data.Helpers
{
	public class EventFlagHelper
	{
		public int ID = 0;

		public bool State = false;
		public void ApplyFlag()
		{
			DSLua.Expr($"SetEventFlag({ID}, {State.ToString().ToLower()})");
		}

		public static void ApplyAll(int[] flagList)
		{
			foreach (int flag_loopVariable in flagList) {
				flag = flag_loopVariable;
				dynamic fuckVb = new EventFlagHelper(flag);
				fuckVb.ApplyFlag();
			}
		}

		public EventFlagHelper(int id, bool state)
		{
			this.ID = id;
			this.State = state;
		}

		public EventFlagHelper(int id)
		{
			State = (id > 0);
			this.ID = Math.Abs(id);
		}

		public static List<EventFlagHelper> GetList(params int[] flags)
		{
			return flags.Select(x => new EventFlagHelper(x)).ToList();
		}

	}
}

File: DaS.ScriptLib.CS/Game/Data/Helpers/EventFlagHelper.vb
Imports DaS.ScriptLib.LuaScripting

Namespace Game.Data.Helpers
    Public Class EventFlagHelper
        Public ID As Integer = 0
        Public State As Boolean = False

        Public Sub ApplyFlag()
            DSLua.Expr($"SetEventFlag({ID}, {State.ToString().ToLower()})")
        End Sub

        Public Shared Sub ApplyAll(flagList As Integer())
            For Each flag In flagList
                Dim fuckVb = New EventFlagHelper(flag)
                fuckVb.ApplyFlag()
            Next
        End Sub

        Public Sub New(id As Integer, state As Boolean)
            Me.ID = id
            Me.State = state
        End Sub

        Public Sub New(id As Integer)
            State = (id > 0)
            Me.ID = Math.Abs(id)
        End Sub

        Public Shared Function GetList(ParamArray flags As Integer()) As List(Of EventFlagHelper)
            Return flags.Select(Function(x) New EventFlagHelper(x)).ToList()
        End Function

    End Class
End Namespace